### DISCUSSION TITLE
Should we remove species properties (ex. diffusivity) from Phase object?

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [PMEAL/OpenPNM#2685] [Development]
I think we should remove species properties such as `pore.diffusivity` from `Phase` objects. The reason is that it's very ambiguous. Of course, for binary mixtures, D12 = D21, but still, it's ambiguous for `Phase` object to have such property. It might make sense for a `BinaryMixture` class to have `pore.diffusivity` as a compromise.

I propose the following:

- Remove species properties from `Phase` objects altogether, and by default use `Mixture` class for mass transport problems.
- Depracate the use of `pore.propname` for species properties altogether, and use `pore.propname.X` instead where X is the species name.

PS. For backwards compatibility, we could still support `pore.diffusivity`, and eventually get rid of it in v4.

**Comment by jgostick** [comment]
I agree completely, but the solution results in the code being significantly more confusing to use, so we remove one source of confusion, but add another.  This is why we're stuck with what we have.  

Incidentally, the new 'Phase properties' functions and classes to support your proposed approach.  But this would require everyone using mixtures, which is more complicated.  

Anyway, definitely worth discussion, and I'd love to hear from active users their thoughts on this thread too.
